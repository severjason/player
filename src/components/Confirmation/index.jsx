// @flow
import * as React from 'react';
import ConfirmationStyle from './style';
import { MdClear, MdCheck } from 'react-icons/lib/md'

type Props = {
  showConfirmation: boolean,
  toggleConfirmation: () => boolean,
  success: () => void,
}

const Confirmation = (props: Props) => (
  <ConfirmationStyle
    className={props.showConfirmation ? 'visible' : ''}
  >
    <div className="item center">
      Logout?
    </div>
    <div className="item">
      <div
        className="button"
        onClick={props.toggleConfirmation}
      >
        <MdClear/>
      </div>
      <div
        className="button"
        onClick={() => {
          props.success();
          props.toggleConfirmation();
        }}>
        <MdCheck/>
      </div>
    </div>

  </ConfirmationStyle>
);


export default Confirmation;
