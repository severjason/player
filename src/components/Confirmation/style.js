import styled from 'styled-components';

const ConfirmationStyle = styled.div`
  position: absolute;
  display: none;
  width: 100%;
  height: 150%;
  max-height: 35rem;
  margin: -.5rem;
  border-radius: 3px;
  background-color: ${props => props.theme.mainColor};
  z-index: 1000;
  &.visible {
    display: block;
  }
  .item {
    display: flex;
    justify-content: center;
    padding: .5rem;
    text-align: center;
    font-size: 1.2rem;
    
    .button {
      font-size: 3rem;
      line-height: 4rem;
      padding: 1rem;
      cursor: pointer;
      
      &:hover {
        color: ${props => props.theme.secondaryColor}
      }
      &:active {
        transform: scale(.9);
      }
    }
    
  }
`;

export default ConfirmationStyle;
