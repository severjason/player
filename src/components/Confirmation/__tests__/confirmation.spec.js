// @flow
import React from 'react';
import Confirmation from '../index';
import renderer from 'react-test-renderer';

type Props = {
  showConfirmation: boolean,
  toggleConfirmation: () => boolean,
  confirmed: () => void,
}

const props: Props = {
  showConfirmation: false,
  toggleConfirmation: () => false,
  confirmed: () => {},
};

it ('renders Confirmation component correctly', () => {
  const tree  = renderer.create(
    <Confirmation
      showConfirmation={props.showConfirmation}
      toggleConfirmation={props.toggleConfirmation}
      confirmed={props.confirmed}
    />).toJSON();
  expect(tree).toMatchSnapshot();
});
