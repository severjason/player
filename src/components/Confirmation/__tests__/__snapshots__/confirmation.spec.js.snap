// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Confirmation component correctly 1`] = `
<div
  className="sc-bdVaJa dwIwhc"
>
  <div
    className="item center"
  >
    Logout?
  </div>
  <div
    className="item"
  >
    <div
      className="button"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m31.6 10.7l-9.3 9.3 9.3 9.3-2.3 2.3-9.3-9.3-9.3 9.3-2.3-2.3 9.3-9.3-9.3-9.3 2.3-2.3 9.3 9.3 9.3-9.3z"
          />
        </g>
      </svg>
    </div>
    <div
      className="button"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m15 27l17.7-17.7 2.3 2.3-20 20-9.3-9.3 2.3-2.3z"
          />
        </g>
      </svg>
    </div>
  </div>
</div>
`;
